EEP 78: Multi-valued comprehensions

michalmuskala opened the pull request:
This is a PR implementing proposal from https://github.com/erlang/eep/pull/75

Maria-12648430 commented on lib/stdlib/test/erl_eval_SUITE.erl:
```diff
@@ -440,6 +442,28 @@ zmc(Config) when is_list(Config) ->
         {bad_generators,{[1], a, #{b=>3}}}),
     ok.
 
+%% EEP 77: multi-comprehensions
```
Shouldn't it be "EEP 78"?

michalmuskala commented:
I rebased the PR and implemented missing parts: erl_syntax, debugger, erl_prettypr, some documentation.
I modeled the changes after https://github.com/erlang/otp/pull/8926 to understand which parts need to change.
